src/Microsoft.ML.DataView/TypeManager.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Microsoft.ML.Internal.DataView;
+
+namespace Microsoft.ML.Data
+{
+    public static class TypeManager
+    {
+        // Types have been used in ML.NET type systems. They can have multiple-to-one type mapping.
+        // For example, UInt32 and Key can be mapped to uint. We enforce one-to-one mapping for all
+        // user-registered types.
+        private static HashSet<Type> _notAllowedRawTypes;
+        private static ConcurrentDictionary<Type, DataViewType> _rawTypeToDataViewTypeMap;
+        private static ConcurrentDictionary<DataViewType, Type> _dataViewTypeToRawTypeMap;
+
+        /// <summary>
+        /// Constructor to initialize type mappings.
+        /// </summary>
+        static TypeManager()
```

Comment:
>TypeManager [](start = 15, length = 11)

IMO 'Manager' is too broad. This type is only managing mappings, and it's just a dictionary with special logic around adding keys. So the following changes might make things clearer:

1) Rename it to DataViewTypeMappings
2) Rename Register() to AddMapping() or Add()
3) Change GetDataViewTypeOrNull() and the other one to indexers, so the code changes to DataViewTypeMappings[someType]

Reply:
`DataViewTypeMappings` looks even more strange to me... This is type register and this is how ONNXMLTools being used.

---
In reply to: [285266926](https://github.com/dotnet/machinelearning/pull/3263#discussion_r285266926) [](ancestors = 285266926)